halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.model.v1;
+
+import com.netflix.spinnaker.halyard.validate.v1.ValidateField;
+import com.netflix.spinnaker.halyard.validate.v1.Validator;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class Updateable<T> {
+  /* Attempts to update field to value, but will be rejected if the validation based on `context fails */
+  public List<String> update(String fieldName, Object value)
+      throws NoSuchFieldException, IllegalAccessException {
+    Field field = this.getClass().getDeclaredField(fieldName);
+    field.setAccessible(true);
+    Object oldValue = field.get(this);
+    field.set(this, value);
+
+    List<Annotation> annotations = Arrays.asList(field.getDeclaredAnnotations());
+    List<String> errors = annotations.stream()
```

Comment:
You could skip directly to this with `Arrays.stream(field.getDeclaredAnnotations())`

https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#stream-T:A-